lib/error_tracker/plugins/pruner.ex

```diff
@@ -0,0 +1,130 @@
+defmodule ErrorTracker.Plugins.Pruner do
+  @moduledoc """
+  Periodically delete resolved errors based on their age.
+
+  Pruning allows you to keep your database size under control by removing old errors that are not
+  needed anymore.
+
+  ## Using the pruner
+
+  To enable the pruner you must register the plugin in the ErrorTracker configuration. This will use
+  the default options, which is to prune errors resolved after 5 minutes.
+
+      config :error_tracker,
+        plugins: [ErrorTracker.Plugins.Pruner]
+
+  You can override the default options by passing them as an argument when registering the plugin.
+
+      config :error_tracker,
+        plugins: [{ErrorTracker.Plugins.Pruner, max_age: :timer.minutes(30)}]
+
+  ## Options
+
+  - `:limit`  - the maximum number of errors to prune on each execution. Occurrences are removed
```

Comment:
1000 errors by default can potentially be very massive. If each error has 10 occurrences, it would be `11,000` deletes in two queries.

It may make more sense to allow setting how many elements to delete per batch, and execute all needed batches per execution of the plugin (? I´m not sure about this last part).

Or even better:

1. Get resolved errors that match the options
2. Iterate over them, one by one, and delete occurrences in batches of ¿1000? elements.
3. Once there is not occurrences left, delete de error and move to the next one.
4. Once all matching errors are deleted, the execution of the plugin ends.

Reply:
Yes, defaults are open for discussion. We could just lower the limit to 200 errors or something like that, and potentially even increase the pruning frequency.

Currently the pruner runs every 30s by default which would mean that each hour it would prune 400 resolved errors. I don't think many teams would resolve so many errors so quickly so, IMHO, it would be a good compromise (especially since anyone can modify the defaults).

Reply:
I didn't have time to check the query plans yet but I believe that with the proper indexes we can rely on the DELETE CASCADE for removing the error's occurrences and avoid complicating the code. At least for now.

Reply:
I dont know if the query planners got better, but the last time I checked if you do a DELETE CASCADE it executed a DELETE on occurrences for each error - not on the same operation as if you do it by code.

Reply:
Confirmed [here](https://dba.stackexchange.com/questions/342027/how-to-know-wether-a-delete-cascade-will-delete-child-records).

I thought that the query planner would be intelligent enough to produce a single delete on the associated table. But it seems that it performs a single delete for each parent row, so deleting 100 parent records would produce 100 delete queries on the child table.